src/sessions/p2p_session.rs

```diff
@@ -900,10 +908,15 @@ impl<T: Config> P2PSession<T> {
     fn check_checksum_send_interval(&mut self) {
         match self.desync_detection {
             DesyncDetection::On { interval } => {
-                let frame_to_send = self.sync_layer.last_saved_frame() - 1;
-                let current = self.current_frame();
-
-                if current % interval as i32 == 0 && frame_to_send > self.max_prediction as i32 {
+                let frame_to_send = if self.last_sent_checksum_frame == NULL_FRAME {
+                    interval as i32
+                } else {
+                    self.last_sent_checksum_frame + interval as i32
+                };
+
+                if frame_to_send <= self.sync_layer.last_confirmed_frame()
+                    && frame_to_send < self.sync_layer.last_saved_frame()
```

Comment:
I'm not completely sure why `<` is needed... I'd think `<=` would be correct, but for some reason that panics. Maybe I misunderstand something about the sync layer... perhaps @gschup has some insights?